src/native/eventpipe/ep-thread.c

```diff
@@ -375,6 +375,13 @@ ep_thread_session_state_alloc (
 	ep_exit_error_handler ();
 }
 
+uint32_t
+ep_thread_session_state_get_buffer_count_estimate(const EventPipeThreadSessionState *thread_session_state)
+{
+	// this is specifically unprotected and allowed to be incorrect due to memory ordering
+	return thread_session_state->buffer_list == NULL ? 0 : thread_session_state->buffer_list->buffer_count;
```

Comment:
Good idea. We've been bitten in the past with things like this. I'll make a more robust comment that details the reasoning/justification.

Reply:
> https://www.pella.com/ideas/doors/entry-doors/fiberglass/glass/

I have a few patches locally that have some `fprintf`s I intend to convert to stress log calls, but they aren't in this PR since it reaches further than this change.